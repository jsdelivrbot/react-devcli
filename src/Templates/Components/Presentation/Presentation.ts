export const statelessJS = (): string => {
    return [
        'import React from \'react\';',
        '',
        'export const REACT__COMPONENT__NAME = (props) => (',
        '   <div>',
        '       REACT__COMPONENT__NAME Stateless Component',
        '   </div>',
        ');',
    ].join('\n');
};

export const statefulJS = (): string => {
    return [
        'import React from \'react\';',
        '',
        'export class REACT__COMPONENT__NAME extends React.Component {',
        '   constructor(props) {',
        '       super(props);',
        '   }',
        '',
        '   render() {',
        '   }',
        '};',
    ].join('\n');
};

export const statelessTS = (): string => {
    return [
        'import React from \'react\';',
        '',
        'export const REACT__COMPONENT__NAME = (props) => (',
        '   <div>',
        '       REACT__COMPONENT__NAME Stateless Component',
        '   </div>',
        ');',
    ].join('\n');
};

export const statefulTS = (): string => {
    return [
        'import React from \'react\';',
        '',
        'export class REACT__COMPONENT__NAME extends React.Component {',
        '   constructor(props) {',
        '       super(props);',
        '   }',
        '',
        '   render() {',
        '   }',
        '};',
    ].join('\n');
};
