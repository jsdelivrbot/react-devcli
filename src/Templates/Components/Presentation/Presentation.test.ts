import {statefulJS, statelessJS, statefulTS, statelessTS} from './Presentation';

describe('Presentation component template test suite', (): void => {
    test('should match stateful presentation component template snapshot for JS language', (): void => {
        expect(statefulJS()).toMatchSnapshot();
    });

    test('should match stateless presentation component template snapshot for JS language', (): void => {
        expect(statelessJS()).toMatchSnapshot();
    });

    test('should match stateful presentation component template snapshot for TS language', (): void => {
        expect(statefulTS()).toMatchSnapshot();
    });

    test('should match stateless presentation component template snapshot for TS language', (): void => {
        expect(statelessTS()).toMatchSnapshot();
    });
});
