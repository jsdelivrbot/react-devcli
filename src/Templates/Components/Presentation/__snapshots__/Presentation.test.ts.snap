// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presentation component template test suite should match stateful presentation component template snapshot for JS language 1`] = `
"import React from 'react';

export class REACT__COMPONENT__NAME extends React.Component {
   constructor(props) {
       super(props);
   }

   render() {
   }
};"
`;

exports[`Presentation component template test suite should match stateful presentation component template snapshot for TS language 1`] = `
"import React from 'react';

export class REACT__COMPONENT__NAME extends React.Component {
   constructor(props) {
       super(props);
   }

   render() {
   }
};"
`;

exports[`Presentation component template test suite should match stateless presentation component template snapshot for JS language 1`] = `
"import React from 'react';

export const REACT__COMPONENT__NAME = (props) => (
   <div>
       REACT__COMPONENT__NAME Stateless Component
   </div>
);"
`;

exports[`Presentation component template test suite should match stateless presentation component template snapshot for TS language 1`] = `
"import React from 'react';

export const REACT__COMPONENT__NAME = (props) => (
   <div>
       REACT__COMPONENT__NAME Stateless Component
   </div>
);"
`;
